### ISSUE TITLE
DotNet TestAccSql failing

### ISSUE THREAD

**Issue by stack72** [pulumi/pulumi-azure-native#1465]
```
   +  azure-native:sql:Database testdb creating error: resource partially created but read failed autorest/azure: Service returned an error. Status=404 Code="ResourceNotFound" Message="The requested resource of type 'Microsoft.Sql/servers/databases' with name 'testdb' was not found.": Code="InternalServerError" Message="An unexpected error occured while processing the request. Tracking ID: 'c100c46b-cb74-439b-80b6-fbf6ba293d48'"
   +  azure-native:sql:Database testdb **creating failed** error: resource partially created but read failed autorest/azure: Service returned an error. Status=404 Code="ResourceNotFound" Message="The requested resource of type 'Microsoft.Sql/servers/databases' with name 'testdb' was not found.": Code="InternalServerError" Message="An unexpected error occured while processing the request. Tracking ID: 'c100c46b-cb74-439b-80b6-fbf6ba293d48'"
   +  azure-native:sql:ServerAzureADOnlyAuthentication ad-only-auth created 
   +  azure-native:network:PrivateEndpoint endpoint created 
   +  pulumi:pulumi:Stack cs-sql-p-it-fv-az201-2-cs-sql-7fefd7e0 creating error: update failed
   +  pulumi:pulumi:Stack cs-sql-p-it-fv-az201-2-cs-sql-7fefd7e0 **creating failed** 1 error
   
  Diagnostics:
    azure-native:sql:Database (testdb):
      error: resource partially created but read failed autorest/azure: Service returned an error. Status=404 Code="ResourceNotFound" Message="The requested resource of type 'Microsoft.Sql/servers/databases' with name 'testdb' was not found.": Code="InternalServerError" Message="An unexpected error occured while processing the request. Tracking ID: 'c100c46b-cb74-439b-80b6-fbf6ba293d48'"
   
    pulumi:pulumi:Stack (cs-sql-p-it-fv-az201-2-cs-sql-7fefd7e0):
      error: update failed
   
  Resources:
      + 12 created
  
  Duration: 2m0s
```

**Comment by danielrbradley** [comment]
Note: this occurs when upgrading `pulumi/pulumi` to `v3.23.0` or `v3.23.03`. Doesn't fail with current `v3.20.0` version.

**Comment by mikhailshilkov** [comment]
> Note: this occurs when upgrading pulumi/pulumi to v3.23.0 or v3.23.03. Doesn't fail with current v3.20.0 version.

That is surprising. Any ideas how this could influence the `InternalServerError` or missing database?

**Comment by danielrbradley** [comment]
No idea - seen same failure 3 times consecutively - and no failures on master without the upgrade.

**Comment by danielrbradley** [comment]
We've now seen the failure on the old version too.

**Comment by danielrbradley** [comment]
The actual failure message is a little misleading. 
1. The SQL Service instance is created.
2. The DB PUT fails with 500 "Internal Server Error"
3. We then try to read the resource in case it was partly created.
4. This fails with a 404 because it wasn't created.

Two possible actions we could do here:
- If the read fails with a 404 we can safely assume nothing was created and just return the original error rather than a partial create.
- We should be able to disable deleting of stacks if the test failed so that we can inspect the created resource group and raise support tickets for any service issues for investigation.

**Comment by danielrbradley** [comment]
Closed by #1471
